src/utils/spark.py

```diff
@@ -173,7 +174,8 @@ def get_filter(self) -> str | None:
         """
         filter = []
         if self.taxyr:
-            filter.append(f"taxyr IN ({', '.join(map(str, self.taxyr))})")
+            min_year, max_year = self.taxyr[0], self.taxyr[-1]
+            filter.append(f"taxyr BETWEEN {min_year} AND {max_year}")
```

Comment:
Using BETWEEN instead of IN makes for cleaner logs, is probably faster, and yields the same results.